src/_pytest/pytester.py

```diff
@@ -1046,23 +1046,24 @@ def collect_by_name(self, modcol, name):
     def popen(self, cmdargs, stdout, stderr, **kw):
         """Invoke subprocess.Popen.
 
-        This calls subprocess.Popen making sure the current working directory
+        This calls subprocess.Popen, making sure the current working directory
         is in the PYTHONPATH.
 
         You probably want to use :py:meth:`run` instead.
-
         """
         env = os.environ.copy()
-        env["PYTHONPATH"] = os.pathsep.join(
-            filter(None, [os.getcwd(), env.get("PYTHONPATH", "")])
-        )
+        env_update = kw.get("env", {})
+        if "PYTHONPATH" not in env_update:
+            env_update["PYTHONPATH"] = os.pathsep.join(
+                filter(None, [os.getcwd(), env.get("PYTHONPATH", "")])
+            )
+        env.update(env_update)
```

Comment:
I came across this again (to pass `$LINES` and `$COLUMNS` for a pty), and think it might be better to use "env" completely, if provided, but copy only the global env otherwise.  I.e. the caller is responsible for copying it then.
On top, "PYTHONPATH" would still be handled always like currently - although I I can imagine someone wanting to not have it adjust/set..
Makes sense?

Reply:
Updated.